src/main/java/edu/ksu/canvas/model/AuthenticationLog.java

```diff
@@ -0,0 +1,117 @@
+package edu.ksu.canvas.model;
+
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.List;
+
+/**
+ * Class to represent Canvas user authentication logs.
+ * See the <a href="https://canvas.instructure.com/doc/api/authentications_log.html">Authentications Log API</a> documentation.
+ * NOTE: The fields returned from Canvas don't match the current documentation
+ */
+public class AuthenticationLog extends BaseCanvasModel implements Serializable {
+    public static final long serialVersionUID = 1L;
+
+    private List<AuthenticationEvent> events;
+    private Object linked;
+    private Object links;
+
+    public enum EventType { login, logout }
+
+    public class AuthenticationEvent {
+
+        private Integer id;
+        private EventType eventType;
+        private AuthenticationEventLinks links;
+
+        private Instant createdAt;
+
+        public Integer getId() {
+            return id;
+        }
+
+        public void setId(Integer id) {
+            this.id = id;
+        }
+
+        public EventType getEventType() {
+            return eventType;
+        }
+
+        public void setEventType(EventType eventType) {
+            this.eventType = eventType;
+        }
+
+        public Instant getCreatedAt() {
+            return createdAt;
+        }
+
+        public void setCreatedAt(Instant createdAt) {
+            this.createdAt = createdAt;
+        }
+
+        public AuthenticationEventLinks getLinks() {
+            return links;
+        }
+
+        public void setLinks(AuthenticationEventLinks links) {
+            this.links = links;
+        }
+
+        public class AuthenticationEventLinks {
+
+            private Integer user;
+            private Integer account;
+            private Integer login;
```

Comment:
These should all be Long, in some situations Canvas will return full IDs that overflow Integers.